pandas/core/dtypes/cast.py

```diff
@@ -1044,7 +1044,7 @@ def maybe_cast_to_datetime(value, dtype, errors="raise"):
                     value = [value]
 
             elif is_timedelta64 and not is_dtype_equal(dtype, _TD_DTYPE):
-                if dtype.name in ("timedelta64", "timedelta64[ns]"):
+                if dtype.name in ("timedelta64", "timedelta64[ns]", "timedelta64[D]"):
```

Comment:
```dtype.kind == 'm'```

Reply:
Since `pandas` doesn't support to convert [`datetime[ps]`](https://github.com/pandas-dev/pandas/blob/master/pandas/tests/series/test_constructors.py#L1352), I used `(dtype.kind == 'm') and (dtype.name != 'timedelta64[ps]')`